### PR TITLE
Support Pact Events

### PR DESCRIPTION
**Pull request by slpopejoy**
Forking change to support Pact Events (https://github.com/kadena-io/pact/pull/819) in output for SPV proofs etc.

TODOs: 
- [ ] Test against mainnet history (current block numbers should be high enough for this)
- [x] Populate block numbers for switch
- [x] merge `kpkgs` pact 3.6 bump to master https://github.com/kadena-io/kpkgs/pull/28 (waiting for mainnet replay to succeed for this)

### CODE REVIEW

File: src/Chainweb/Version.hs [hunk 1]
```diff
@@ -839,6 +840,13 @@ enableModuleNameFix2 Mainnet01 bh = bh >= 752214 -- ~ 2020-07-17 0:00:00 UTC
 enableModuleNameFix2 Testnet04 bh = bh >= 289966 -- ~ 2020-07-13
 enableModuleNameFix2 _ bh = bh >= 2
 
+-- | Enable serialization of Pact Events (Pact 3.6) in output
+enablePactEvents :: ChainwebVersion -> BlockHeight -> Bool
+enablePactEvents Mainnet01 bh = bh >= 1_000_000 -- TODO
+enablePactEvents Testnet04 bh = bh >= 600_000 -- TODO
```
Comment (emilypi) [inline src/Chainweb/Version.hs:846]: These are still todo?

Comment (slpopejoy) [reply src/Chainweb/Version.hs:846]: Yes

**Review by emilypi — APPROVED** [review]

File: src/Chainweb/Version.hs [hunk 2]
```diff
@@ -839,6 +840,13 @@ enableModuleNameFix2 Mainnet01 bh = bh >= 752214 -- ~ 2020-07-17 0:00:00 UTC
 enableModuleNameFix2 Testnet04 bh = bh >= 289966 -- ~ 2020-07-13
 enableModuleNameFix2 _ bh = bh >= 2
 
+-- | Enable serialization of Pact Events (Pact 3.6) in output
+enablePactEvents :: ChainwebVersion -> BlockHeight -> Bool
+enablePactEvents Mainnet01 bh = bh >= 1_013_500 -- 2020-10-15 18:47:13 UTC
```
Comment (slpopejoy) [inline src/Chainweb/Version.hs:845]: @larskuhtz @mightybyte please confirm

File: src/Chainweb/Version.hs [hunk 3]
```diff
@@ -839,6 +840,13 @@ enableModuleNameFix2 Mainnet01 bh = bh >= 752214 -- ~ 2020-07-17 0:00:00 UTC
...
+enablePactEvents Testnet04 bh = bh >= 558_000 -- 2020-10-15 17:11:45 UTC
```
Comment (slpopejoy) [inline src/Chainweb/Version.hs:846]: @larskuhtz please confirm

File: src/Chainweb/Version.hs (hunk 2 above)
Comment (mightybyte) [reply src/Chainweb/Version.hs:845]: Passes my rough sanity check.  @larskuhtz should probably put it into his more detailed calculation.

File: src/Chainweb/Version.hs (hunk 3 above)
Comment (larskuhtz) [reply src/Chainweb/Version.hs:846]: 👍 

this is what my script gives me:

```
mainnet: 2020-10-15 18:46:15
testnet: 2020-10-15 17:11:14
```

**Review by larskuhtz — APPROVED** [review]

**Pull request merged**